hack/ci/e2e.sh

```diff
@@ -106,8 +111,33 @@ apiVersion: kind.sigs.k8s.io/v1alpha3
 nodes:
 # the control plane node
 - role: control-plane
+  extraMounts:
+  - hostPath: "${ARTIFACTS}/logs/apiserver-audit.log"
+    containerPath: /var/log/apiserver-audit.log
+  - hostPath: /tmp/audit-policy.yaml
+    containerPath: /etc/kubernetes/audit-policy.yaml
 - role: worker
-- role: worker
+  replicas: 2
+kubeadmConfigPatches:
+- |
+  apiVersion: kubeadm.k8s.io/v1beta1
```

Comment:
oh, so v1beta1 is available in 1.13, 1.14 and master (1.15)
i think for 1.11 and 1.12 v1alpha2 has to be used:
https://github.com/kubernetes-sigs/kind/blob/6c35d5e269d0844732f026e8a47c4bf14421dffd/pkg/cluster/internal/kubeadm/config.go
https://github.com/kubernetes/kubernetes/blob/release-1.12/cmd/kubeadm/app/apis/kubeadm/v1alpha2/types.go

```

Reply:
i think what happens here is that the CI jobs for the old versions, don't apply this patch.
extra patches for them need to be added:
```
kubeadmConfigPatches:
- |
   ... patch for v1beta1 ClusterConfiguration
- |
  .... patch for v1alpha2 InitConfiguration
```

Reply:
also ideally the 1.11 job should be removed at this point.
1.11 is out of support project wide.

Reply:
I'm removing 1.11 over at https://github.com/kubernetes/test-infra/pull/12387

Reply:
if 1.11 is removed for 1.12 kubeadm's v1alpha3 can be used.
please mind that i'm not sure if @BenTheElder wants to keep 1.11 for longer.

Reply:
i'd like to try to support 1.11 nominally, but 1.11 testing did have to be removed as we can't get e2e.test patched for some bugs we hit :(

Reply:
I was trying to use this PR as a basis to get audit working for a `kind` cluster based off current release, and it seems I had to use `apiVersion: kubeadm.k8s.io/v1beta2` for `kubeadmConfigPatches`, just FYI

Reply:
right, 1.15+ is v1beta2

Reply:
to clarify, kind patches do not support kubeadm v1beta1 if you targeting 1.15, and only support v1beta2. v1beta2 is the current preferred type in kubeadm, but v1beta1 will continue to be supported until 1.18.

kubeadm will soon enter a state where it will have 3 beta versions supported at once.